ckan/lib/uploader.py

```diff
@@ -98,44 +74,30 @@ def get_max_resource_size() -> int:
 
 
 class Upload(object):
-    storage_path: Optional[str]
+    storage: Optional[files.Storage]
     filename: Optional[str]
-    filepath: Optional[str]
     object_type: Optional[str]
     old_filename: Optional[str]
-    old_filepath: Optional[str]
-    upload_file: Optional[IO[bytes]]
+    upload_file: files.Upload | None
 
     def __init__(self,
                  object_type: str,
                  old_filename: Optional[str] = None) -> None:
         ''' Setup upload by creating a subdirectory of the storage directory
         of name object_type. old_filename is the name of the file in the url
         field last time'''
-
-        self.storage_path = None
         self.filename = None
-        self.filepath = None
-        path = get_storage_path()
-        if not path:
-            return
-        self.storage_path = os.path.join(path, 'storage',
-                                         'uploads', object_type)
-        # check if the storage directory is already created by
-        # the user or third-party
-        if os.path.isdir(self.storage_path):
-            pass
-        else:
-            try:
-                os.makedirs(self.storage_path)
-            except OSError as e:
-                # errno 17 is file already exists
-                if e.errno != 17:
-                    raise
         self.object_type = object_type
         self.old_filename = old_filename
-        if old_filename:
-            self.old_filepath = os.path.join(self.storage_path, old_filename)
+
+        storage_name = f"{object_type}_uploads"
+        self.storage = files.storages.get(storage_name)
+        if not self.storage:
+            log.debug(
```

Comment:
If the storage for this type is not configured and something is trying to upload stuff perhaps we should use `log.warning()` or higher?